collector/handler.go

```diff
@@ -0,0 +1,139 @@
+package collector
+
+import (
+	"net/http"
+	"regexp"
+
+	"github.com/apex/log"
+	"github.com/gin-gonic/gin"
+)
+
+// CreateReportReq is a client request for the POST /report API endpoint
+type CreateReportReq struct {
+	SoftwareName      string `json:"software_name"`
+	SoftwareVersion   string `json:"software_version"`
+	ProbeASN          string `json:"probe_asn"`
+	ProbeCC           string `json:"probe_cc"`
+	TestName          string `json:"test_name"`
+	TestVersion       string `json:"test_version"`
+	DataFormatVersion string `json:"data_format_version"`
+	Format            string `json:"format"`
+	// The below fields are optional
+	TestStartTime string   `json:"test_start_time,omitempty"`
+	InputHashes   []string `json:"input_hashes,omitempty"`
+	TestHelper    string   `json:"test_helper,omitempty"`
+	Content       string   `json:"content,omitempty"`
+	ProbeIP       string   `json:"probe_ip,omitempty"`
+}
+
+var supportedFormats = []string{"json"}
+var softwareNameRegexp = regexp.MustCompile("^[0-9A-Za-z_\\.+-]+$")
+var testNameRegexp = regexp.MustCompile("^[a-zA-Z0-9_\\- ]+$")
+var probeASNRegexp = regexp.MustCompile("^AS[0-9]{1,10}$")
+var probeCCRegexp = regexp.MustCompile("^[A-Z]{2}$")
+
+// CreateReportHandler for report creation
+func CreateReportHandler(c *gin.Context) {
+	store := c.MustGet("Storage").(*Storage)
+
+	var req CreateReportReq
+
+	if err := c.BindJSON(&req); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	if req.Format == "" {
+		req.Format = supportedFormats[0]
+	}
+	if stringInSlice(req.Format, supportedFormats) != true {
+		c.JSON(http.StatusNotAcceptable, gin.H{"error": "invalid format. Must be either json or yaml"})
+		return
+	}
+
+	reportID, err := CreateNewReport(store, req.Format)
+	if err != nil {
+		// XXX check this against the spec
+		c.JSON(http.StatusNotAcceptable, gin.H{
+			"error": "invalid request",
+		})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{
+		"backend_version":   Version,
+		"report_id":         reportID,
+		"supported_formats": supportedFormats,
+	})
+	return
+}
+
+// DeprecatedUpdateReportHandler is for legacy clients
+func DeprecatedUpdateReportHandler(c *gin.Context) {
+	c.JSON(http.StatusOK, gin.H{
```

Comment:
Thanks for pointing this out. I need to double check this, I suspect it may be that since we don't support YAML there is no point in having this here, but if that is the case I agree we should drop this code and not pretend it works.

Reply:
As discussed in https://github.com/ooni/collector/pull/3#issuecomment-501759144, we'll drop this code.